rfd/0027-account-lifecycle.md

```diff
@@ -0,0 +1,195 @@
+screen with a message:
+
+```
+Please save these account recovery tokens in a safe offline place.
+You can use each token once if you loose your second factor.
+
+* recover-example-teleport-sh-token-1
+* recover-example-teleport-sh-token-2
+* recover-example-teleport-sh-token-3
+
+```
+
+The cloud prefix is for customer convenience in case if they forgot the domain
+name as well.
+
+### Recovery scenarios
+
+The first two recovery scenarios work for every system user, reducing the load
+on Teleport system administrators as well.
+
+**User lost a second factor**
+
+If user lost a second factor, but not password, they can recover using
+recovery token:
+
+* Login with a username and password
+* Instead of pressing second factor, enter a one-time recovery token
+* Enroll new second factor right away.
+
+The UI flow should prompt a blocking request for a second factor right
+after user has entered a valid recovery token.
+
+* The used account recovery token is no longer valid. The new recovery tokens are printed to the user.
+
+This flow ensures two factors are verified: the valid password and a recovery token.
+
+**User lost a password**
+
+If user lost a password, but not a second factor, they could recover using
+a recovery link sent to their email and a second factor:
+
+* User enters their email and clicks forgot password.
+* The recovery system sends a short-lived reset short lived password link to the email and
+prints (in all cases, whether the email is valid or invalid):
+
+`Please check your inbox for a recovery link and follow the instructions.`
+
+The link includes a crypto-random short-lived 32 byte token.
+
+* Once user clicks on the link, they have to present a valid second factor and a
+recovery token, after that they will be able to reset their password.
+
+This flow ensure that two factors are verified:
+
+* Second factor TOTP token or U2F device
+* A valid account recovery token.
+
+Notice that we do not consider a valid email a valid verification factor,
+although it's a nice-to-have protection layer.
+
+* The used account recovery token is no longer valid. The new recovery tokens are printed to the user.
+
+**Internal request for cancelling the account**
+
+To request account cancellation, a user with sufficient privileges should:
+
+* Be logged into Teleport UI.
+* Press a "cancel my account" button in the UI.
+* UI should verify that they own a second factor.
+* Verify that they have write permission on the billing.
+
+```
+We have marked your account as inactive. In 10 days, we are going to delete the account and all it's data.
+To prevent this, log into Teleport and press "revert account cancelation".
+```
+
+If anyone logs into this account, we should show them the warning:
+
+```
+Your account was marked as inactive at the request of `email`.
+To prevent account from being deleted, log into Teleport and press "revert account cancelation".
+```
+
+Only users with write permissions to billing should be able to revert account
```

Comment:
All users with write permissions to billing should get notified via email too.

Reply:
updated